server/connectors/coroot_connector/client.py

```
@@ -0,0 +1,497 @@
+                f"{self.url}/api/login",
+                json={"email": self.email, "password": self.password},
+                timeout=COROOT_TIMEOUT,
+            )
+        except requests.RequestException as exc:
+            logger.error("[COROOT] Login network error: %s", exc)
+            raise CorootAPIError("Unable to reach Coroot server") from exc
+
+        if resp.status_code == 401:
+            raise CorootAPIError(
+                f"Invalid email or password"
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)